lib/trento/settings.ex

```diff
@@ -154,6 +155,44 @@ defmodule Trento.Settings do
     end
   end
 
+  # Analytics settings
+
+  @spec get_analytics_settings() ::
+          {:ok, AnalyticsSettings.t()} | {:error, :analytics_settings_not_configured}
+  def get_analytics_settings do
+    case Repo.one(AnalyticsSettings.base_query()) do
+      %AnalyticsSettings{} = settings -> {:ok, settings}
+      nil -> {:error, :analytics_settings_not_configured}
+    end
+  end
+
+  @spec change_analytics_opt_in(boolean()) ::
+          {:ok, AnalyticsSettings.t()}
+          | {:error, :analytics_settings_not_configured}
+  def change_analytics_opt_in(value) do
+    case Repo.one(AnalyticsSettings.base_query()) do
+      %AnalyticsSettings{} = settings ->
+        {:ok, settings}
+
+        settings
```

Comment:
This update or insert usually is implemented in a different way. Basically, you don't use `update` and you use `insert` with the `on_conflicts` field.

```
 %AnalyticsSettings{}
|> AnalyticsSettings.changeset(%{
    opt_in: value
})
|> Repo.insert(conflict_target: :id, on_conflict, {:replace, [:opt_in]})
```

Anyway, I don't like much the `insert if not found` idea.

Reply:
I have changed this to use the `Repo.insert` approach.

Reply:
Let' use `Repo.update` @jagabomb 
It is more semantically and technically correct.
I just wanted to show you how you implement a ` insert or update` thing, but here we are not going to insert never, so using `update` is better in all the cases